src/opm/input/eclipse/Schedule/Well/WellConnections.cpp

```diff
@@ -17,66 +17,77 @@
 */
 
+#include <opm/input/eclipse/Schedule/Well/WellConnections.hpp>
+
+#include <opm/common/OpmLog/KeywordLocation.hpp>
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/common/utility/ActiveGridCells.hpp>
+
+#include <opm/io/eclipse/rst/connection.hpp>
+
+#include <opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+#include <opm/input/eclipse/EclipseState/Grid/FieldPropsManager.hpp>
+#include <opm/input/eclipse/Schedule/Well/Connection.hpp>
+#include <opm/input/eclipse/Schedule/ScheduleGrid.hpp>
+
+#include <opm/input/eclipse/Units/Units.hpp>
+
+#include <opm/input/eclipse/Deck/DeckRecord.hpp>
+
 #include <algorithm>
+#include <array>
 #include <cassert>
 #include <cmath>
 #include <cstddef>
-#include <iostream>
 #include <limits>
 #include <stdexcept>
 #include <string>
 #include <utility>
-#include <fmt/format.h>
+#include <vector>
 
-#include <opm/common/OpmLog/OpmLog.hpp>
-#include <opm/common/utility/ActiveGridCells.hpp>
-#include <opm/input/eclipse/Units/Units.hpp>
-#include <opm/io/eclipse/rst/connection.hpp>
-#include <opm/common/OpmLog/KeywordLocation.hpp>
-#include <opm/input/eclipse/Deck/DeckRecord.hpp>
-#include <opm/input/eclipse/EclipseState/Grid/EclipseGrid.hpp>
-#include <opm/input/eclipse/EclipseState/Grid/FieldPropsManager.hpp>
-#include <opm/input/eclipse/Schedule/Well/Connection.hpp>
-#include <opm/input/eclipse/Schedule/Well/WellConnections.hpp>
-#include <opm/input/eclipse/Schedule/ScheduleGrid.hpp>
+#include <stddef.h>
 
-namespace Opm {
+#include <fmt/format.h>
 
 namespace {
 
     // Compute direction permutation corresponding to completion's
     // direction.  First two elements of return value are directions
     // perpendicular to completion while last element is direction
     // along completion.
-inline std::array< size_t, 3> directionIndices(const Opm::Connection::Direction direction)
+    std::array<std::size_t, 3>
+    directionIndices(const Opm::Connection::Direction direction)
     {
         switch (direction) {
         case Opm::Connection::Direction::X:
-            return {{ 1,2,0 }};
+            return {{ 1, 2, 0 }};
 
         case Opm::Connection::Direction::Y:
-            return {{ 2,0,1}};
+            return {{ 2, 0, 1}};
 
         case Opm::Connection::Direction::Z:
-            return {{ 0,1,2 }};
+            return {{ 0, 1, 2 }};
         }
-        // All enum values should be handled above. Therefore
-        // we should never reach this one. Anyway for the sake
-        // of reduced warnings we throw an exception.
-        throw std::invalid_argument("unhandled enum value");
 
+        // All enum values should be handled above. Therefore we should
+        // never reach this one. Anyway for the sake of reduced warnings we
+        // throw an exception.
+        throw std::invalid_argument {
+            fmt::format("Unhandled direction enumeration value '{}'",
+                        static_cast<int>(direction))
```

Comment:
Of course this is an improvemnet but is an int really useful to a user? Shouldn't it be  a string.

Reply:
> Shouldn't it be a string.

Good idea.  I've pushed an update that uses `Connection::Direction2String()` instead.